helper/mac/writejob.cpp

```diff
@@ -210,9 +282,9 @@ bool WriteJob::writeCompressed() {
     }
 }
 
-void WriteJob::check() {
-    QFile target("/dev/r"+where);
-    target.open(QIODevice::ReadOnly);
+void WriteJob::check(int fd) {
+    QFile target;
+    target.open(fd, QIODevice::ReadOnly, QFileDevice::AutoCloseHandle);
```

Comment:
But you're using `AutoCloseHandle` here as well, after the previous `QFile` has closed the FD. I think it should be only in this function but not in the write ones.